framework/src/auxkernels/MortarNodalAuxKernel.C

```diff
@@ -78,17 +78,30 @@ MortarNodalAuxKernelTempl<ComputeValueType>::compute()
   ComputeValueType value(0);
   Real total_volume = 0;
 
-  const auto & its = amg().secondariesToMortarSegments(*_current_node);
+  const std::vector<AutomaticMortarGeneration::MortarFilterIter> & its =
+      amg().secondariesToMortarSegments(*_current_node);
   std::array<MortarNodalAuxKernelTempl<ComputeValueType> *, 1> consumers = {{this}};
 
+  std::vector<AutomaticMortarGeneration::MortarFilterIter> iterators;
+  for (size_t i = 0; i < its.size(); i++)
+  {
+    mooseAssert(its[i]->first->active(),
+                "We loop over active elements when building the mortar segment mesh, so we golly "
+                "well hope this is active.");
+
+    if (!its[i]->second.empty())
+      // This is local and the mortar segment set isn't empty, so include
+      iterators.push_back(its[i]);
+  }
```

Comment:
I don't understand this. Do we have cases where there are no mortar segments associated with a secondary? That seems like a problem in the mortar mesh generation or I guess in the generation of this data.

Reply:
As the code stands right now, we have secondary elements with no corresponding mortar segment quadrature points (I assume just corresponding mortar segments). Then we make calls in `MortarUtils.h` like this:

``` if (assembly.needDual() )
      assembly.reinitDual(secondary_face_elem, secondary_xi_pts, JxW);
```
      
which require having some mortar segments associated with secondary elements, otherwise we get exceptions from libMesh. 

> Do we have cases where there are no mortar segments associated with a secondary?

I guess in a general case we can have cases where there are no mortar segment associated with a secondary element (say a circle in contact with a small plane, where the circle is secondary surface).

Reply:
There is probably a better way to deal with this.

Reply:
> I guess in a general case we can have cases where there are no mortar segment associated with a secondary element (say a circle in contact with a small plane, where the circle is secondary surface).

In the early days we would have had mortar segments from these, they just wouldn't have any bisections (loose use of bisection here) by inverse projection of primary nodes onto the secondary face element, e.g. the mortar segment elements would have corresponded exactly to the secondary face elements. However, @jbadger95 may have removed these IIRC, e.g. we only have mortar segment elements for cases in which the secondary projects onto the primary. Perhaps this is why `second` can be empty now?

Reply:
As far as I remember this is correct @lindsayad. There used to be a `has_primary` flag in mortar segments but because there was a lot of confusion around edge dropping we changed it so that only mortar segments with a primary element are added to the mortar segment mesh.

Reply:
@recuero would you want to check and see whether you can eliminate the presence of "empty" secondaries from the data returned by `secondariesToMortarSegments` so that we don't have to do manual manipulation in user code (like in `MortarNodalAuxKernel`)? If you don't have time, I can look into it

Reply:
Yeah, that sounds like the way to go. I can look into that. Give me a bit of time.